Begin mass migration from TestRunner.idl to testRunnerJS

achristensen07 opened the pull request:
#### d50d95fe9f0eaf7059a1dcb5636745027899b6d5
<pre>
Begin mass migration from TestRunner.idl to testRunnerJS
<a href="https://bugs.webkit.org/show_bug.cgi?id=297973">https://bugs.webkit.org/show_bug.cgi?id=297973</a>
<a href="https://rdar.apple.com/159291124">rdar://159291124</a>

Reviewed by Tim Horton.

There are 3 classes of simple things to migrate.  This begins their migration.
1. Notifications implemented with postPageMessage can migrate to just calling post.
2. Synchronous queries implemented with postSynchronousMessage can migrate to being
   asynchronous queries by adding await to the tests that use them.
3. Asynchronous ueries implemented with postMessageWithAsyncReply can migrate to either
   using post and have the test await the result, or if the callback parameter model
   is to be kept then we can make a little wrapper in testRunnerJS that awaits the
   result of post then calls the callback with it.

This gains implementation experience of some primitives needed to migrate other
applications off of the injected bundle by migrating WebKitTestRunner off the
injected bundle.

* LayoutTests/http/wpt/webauthn/public-key-credential-create-failure-local.https.html:
* LayoutTests/http/wpt/webauthn/public-key-credential-create-success-local.https.html:
* Tools/WebKitTestRunner/InjectedBundle/Bindings/TestRunner.idl:
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.cpp:
(WTR::TestRunner::getApplicationManifestThen): Deleted.
(WTR::TestRunner::keyExistsInKeychain): Deleted.
(WTR::TestRunner::waitBeforeFinishingFullscreenExit): Deleted.
(WTR::TestRunner::scrollDuringEnterFullscreen): Deleted.
(WTR::TestRunner::finishFullscreenExit): Deleted.
(WTR::TestRunner::requestExitFullscreenFromUIProcess): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.h:
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::requestExitFullscreenFromUIProcess):
(WTR::adoptAndCallCompletionHandler):
(WTR::if):
(WTR::CompletionHandler&lt;void):
* Tools/WebKitTestRunner/TestController.h:
* Tools/WebKitTestRunner/TestInvocation.cpp:
(WTR::TestInvocation::didReceiveMessageFromInjectedBundle):
(WTR::TestInvocation::didReceiveSynchronousMessageFromInjectedBundle):

Canonical link: <a href="https://commits.webkit.org/299219@main">https://commits.webkit.org/299219@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/4b3c9ef444f33ffe7e3a4a9ea3a1fdb736f65a85

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/118270 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37950 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28593 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124432 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/70318 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/d419e465-9e04-4999-9f4c-24d5d87afb27) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/120148 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38645 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46532 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89756 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/59390 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/34e128af-c38e-4cf6-b439-8909b322d6e2) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/121223 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30782 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/106042 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70249 "Found 1 new API test failure: TestWTF:WTF_Condition.OneProducerOneConsumerOneSlotTimeout (failure)") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/2c1c6677-dc14-4f60-a4aa-333873a8db59) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/29849 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/24160 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/68096 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100209 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/24339 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127505 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/45176 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/34059 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/98439 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45539 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102261 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98225 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43616 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21606 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/41652 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18847 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/45046 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/50722 "Build is in progress. Recent messages:") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44508 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/47853 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/46196 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/4b3c9ef444f33ffe7e3a4a9ea3a1fdb736f65a85. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49945

achristensen07 committed:
Begin mass migration from TestRunner.idl to testRunnerJS
https://bugs.webkit.org/show_bug.cgi?id=297973
rdar://159291124

Reviewed by Tim Horton.

There are 3 classes of simple things to migrate.  This begins their migration.
1. Notifications implemented with postPageMessage can migrate to just calling post.
2. Synchronous queries implemented with postSynchronousMessage can migrate to being
   asynchronous queries by adding await to the tests that use them.
3. Asynchronous ueries implemented with postMessageWithAsyncReply can migrate to either
   using post and have the test await the result, or if the callback parameter model
   is to be kept then we can make a little wrapper in testRunnerJS that awaits the
   result of post then calls the callback with it.

This gains implementation experience of some primitives needed to migrate other
applications off of the injected bundle by migrating WebKitTestRunner off the
injected bundle.

* LayoutTests/http/wpt/webauthn/public-key-credential-create-failure-local.https.html:
* LayoutTests/http/wpt/webauthn/public-key-credential-create-success-local.https.html:
* Tools/WebKitTestRunner/InjectedBundle/Bindings/TestRunner.idl:
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.cpp:
(WTR::TestRunner::getApplicationManifestThen): Deleted.
(WTR::TestRunner::keyExistsInKeychain): Deleted.
(WTR::TestRunner::waitBeforeFinishingFullscreenExit): Deleted.
(WTR::TestRunner::scrollDuringEnterFullscreen): Deleted.
(WTR::TestRunner::finishFullscreenExit): Deleted.
(WTR::TestRunner::requestExitFullscreenFromUIProcess): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.h:
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::requestExitFullscreenFromUIProcess):
(WTR::adoptAndCallCompletionHandler):
(WTR::if):
(WTR::CompletionHandler<void):
* Tools/WebKitTestRunner/TestController.h:
* Tools/WebKitTestRunner/TestInvocation.cpp:
(WTR::TestInvocation::didReceiveMessageFromInjectedBundle):
(WTR::TestInvocation::didReceiveSynchronousMessageFromInjectedBundle):

Canonical link: https://commits.webkit.org/299219@main

webkit-commit-queue commented:
Committed 299219@main (d50d95fe9f0e): <https://commits.webkit.org/299219@main>

Reviewed commits have been landed. Closing PR #49945 and removing active labels.
